index.bs

```diff
@@ -958,49 +958,23 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 				1. [=Queue a global task=] on the [=clipboard task source=], given |realm|'s [=realm/global object=], to perform the below steps:
 
-					1. For each |systemClipboardItem| in |data|:
+					1. For each |systemClipboardItem| of |data|:
 
-						1. For each |systemClipboardRepresentation| in |systemClipboardItem|:
+						1. For each |systemClipboardRepresentation| of |systemClipboardItem|:
 
 							1. Let |mimeType| be the result of running the [=well-known mime type from os specific format=] algorithm given |systemClipboardRepresentation|'s [=system clipboard representation/name=].
 							
-							1. If |mimeType| is null, continue this loop.
+							1. If |mimeType| is null, then [=iteration/continue=].
 
-							1. Let |representation| be a new [=representation=].
+							1. If |mimeType|'s [=MIME type/essence=] is "`text/plain`":
 
-							1. If |representation|'s [=representation/MIME type=] [=MIME type/essence=] is "text/plain", then:
+								1. Resolve |p| with |systemClipboardRepresentation|'s [=system clipboard representation/data=].
```

Comment:
`[=system clipboard representation/data=]` can be a string or a blob, right? But we want to always return a string. How is that going to work now?